src/NuGet.Core/NuGet.Protocol/RemoteRepositories/HttpFileSystemBasedFindPackageByIdResource.cs

```diff
@@ -497,14 +508,27 @@ public class HttpFileSystemBasedFindPackageByIdResource : FindPackageByIdResourc
                         logger,
                         cancellationToken);
                 }
-                catch (Exception ex) when (retry < 3)
+                catch (Exception ex) when (retry < _maxRetries)
                 {
                     var message = string.Format(CultureInfo.CurrentCulture, Strings.Log_RetryingFindPackagesById, nameof(FindPackagesByIdAsync), uri)
                         + Environment.NewLine
                         + ExceptionUtilities.DisplayMessage(ex);
                     logger.LogMinimal(message);
+
+                    if (NuGetEnvironment.EnhancedHttpRetryEnabled &&
+                        ex.InnerException != null &&
+                        ex.InnerException is IOException &&
+                        ex.InnerException.InnerException != null &&
+                        ex.InnerException.InnerException is System.Net.Sockets.SocketException)
+                    {
+                        // https://github.com/NuGet/Home/issues/11027
+                        // An IO Exception with inner SocketException indicates server hangup ("Connection reset by peer").
+                        // Azure DevOps feeds sporadically do this due to mandatory connection cycling.
+                        // Stalling an five extra seconds gives extra time for the hosts to pick up new session tokens if it occurs.
+                        await Task.Delay(5000);
```

Comment:
Let's specify a `TimeSpan` instead (also in `FindPackagesByIdNupkgDownloader.cs`)
```C#
await Task.Delay(TimeSpan.FromSeconds(5));
```